File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.py
def solution(numbers, target):
    global answer
    answer = 0
    # idx : 인덱스, v : 계산 값, s : 수식
    def find(idx, v):
        global answer
        if idx == len(numbers):
            if v == target:
                answer += 1
        else:
            find(idx+1,v+numbers[idx])
            find(idx+1,v-numbers[idx])
    
    find(0,0)
    return answer

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.java
class Solution {
    static int answer = 0;
    
    public int solution(int[] numbers, int target) {
        find(0, 0, numbers, target);
        return answer;
    }

    public void find(int idx, int v, int[] numbers, int target) {
        if (idx == numbers.length) {
            if (v == target) {
                answer++;
            }
        } else {
            find(idx + 1, v + numbers[idx], numbers, target);
            find(idx + 1, v - numbers[idx], numbers, target);
        }
    }
}